internal/shim-sev/src/syscall.rs

```diff
@@ -205,11 +208,29 @@ impl SyscallHandler for Handler {
         &mut self,
         _nonce: UntrustedRef<u8>,
         _nonce_len: libc::size_t,
-        _buf: UntrustedRefMut<u8>,
-        _buf_len: libc::size_t,
+        buf: UntrustedRefMut<u8>,
+        buf_len: libc::size_t,
     ) -> sallyport::Result {
-        self.trace("get_att", 0);
-        Ok([0.into(), SEV_TECH.into()])
+        self.trace("get_attestation", 4);
+
+        match SEV_SECRET.read().deref() {
+            Some(secret) => {
+                let mut result_len = size_of::<SevSecret>();
+
+                if buf_len != 0 {
+                    result_len = result_len.min(buf_len);
+                    let buf = buf.validate_slice(buf_len, self).ok_or(libc::EFAULT)?;
+
+                    buf[..result_len].copy_from_slice(&secret.data[..result_len]);
```

Comment:
Is the expectation here that userspace knows exactly how long the secret is?

Reply:
for now.. yes